tests/performance/test_performance.py

```diff
@@ -0,0 +1,130 @@
+import json
+import logging
+from collections.abc import Callable
+from dataclasses import dataclass
+from functools import partial
+from time import process_time
+
+from databricks.labs.blueprint.parallel import Threads
+from databricks.labs.lsql.backends import SqlBackend
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service import iam
+
+from databricks.labs.ucx.workspace_access.base import Permissions
+from databricks.labs.ucx.workspace_access.groups import MigrationState
+from databricks.labs.ucx.workspace_access.manager import PermissionManager
+
+logger = logging.getLogger(__name__)
+
+NB_OF_TEST_WS_OBJECTS = 100
+
+
+@dataclass
+class WorkspaceObject:
+    fixture: Callable
+    permissions: list[iam.PermissionLevel]
+    id_attribute: str
+    type: str
+
+
+def test_apply_group_permissions_experimental_performance(
+    ws: WorkspaceClient,
+    sql_backend: SqlBackend,
+    inventory_schema,
+    permission_manager: PermissionManager,
+    make_migrated_group,
+    make_experiment,
+    make_model,
+    make_cluster_policy,
+    env_or_skip,
+):
+    # Making sure this test can only be launched from local
+    env_or_skip("PWD")
```

Comment:
```suggestion
    env_or_skip("IDE_PROJECT_ROOTS")
```
isn't $PWD too common?